### PR TITLE
Gate --binary-location native commands on runnability, self-healing the exec bit

### PR DESCRIPTION
**Pull request by brovatten**
## Problem

#354 fixed the `PermissionError [Errno 13]` on a non-executable gopls for the default servers-dir flow, but the `--binary-location` flow bypasses all of it: `bootstrap_environment` calls `update_config(binary_location)` *instead of* `ensure_tools()` (so no install and no chmod self-heal), and `update_command_paths` rewrote native-tool commands (gopls, csharp-ls, rust-analyzer, tokei) to absolute paths under `<binary_location>/bin/<platform>/` with no exists/X_OK check. A binary left at 0644 by a mode-dropping unpack then reached `Popen` as an absolute path and failed with EACCES — and because the path was absolute, `build_config`'s PATH fallback was disabled, so retries failed deterministically. Reproducible on main by pointing `--binary-location` at `~/.codeboarding/servers`, which yields the exact traceback from the original report. The same bypass is reachable via `health_main` and the Pro wrapper's `_apply_binary_location`/serve path.

## Fix

`update_command_paths` now only commits the absolute path when the binary is actually runnable, via a new `_runnable_native()` gate that:

- requires a regular file (`isfile`, not `exists` — a traversable *directory* named like the tool passes an exists+X_OK check and recreates the same EACCES-with-dead-fallback failure)
- restores a lost exec bit in place (`chmod 0o755`), mirroring the `tool_registry.installers` repair, but refuses to chmod through a symlink since `binary_location` is caller-supplied
- on Windows accepts the `.exe` sibling of the suffix-less path that `CreateProcess` resolves
- keeps the bare command name on failure (logged), so `build_config`'s PATH fallback stays available — failure parity with the default servers-dir path

The candidate is re-derived from the bare tool name on every call, so re-invocation re-validates against the current bin dir instead of trusting a stale absolute path from an earlier call.

The helper lives in `vscode_constants.py` (stdlib-only) rather than `tool_registry.paths` because `tool_registry.manifest` imports this module — importing back would be circular.

## Tests

New `TestUpdateCommandPathsNativeGate` covers: executable binary wired absolute (gopls + tokei, i.e. both config sections), exec-bit repair (with log assertion), missing binary keeps the bare name (POSIX and Windows), unrepairable binary keeps the bare name (chmod failure, with warning assertion), directory and symlink impostors rejected, re-invocation after the old location disappears, and the Windows `.exe`-sibling contract. Exec-bit tests are skipped on Windows hosts where chmod can't drop executability.

Full unit suite passes (1612 passed, 28 skipped); black and mypy clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive tests for native command path handling, covering executable binaries, permission repair, missing or non-executable binaries, directory/name collisions, symlink edge cases, reinvocation behavior, and Windows-specific resolution.

* **Improvements**
  * Smarter handling of native command paths: prefer absolute paths only for runnable binaries, fall back to bare command names otherwise, with permission-repair attempts and better Windows compatibility.

### CODE REVIEW

File: vscode_constants.py [hunk 1]
```diff
@@ -12,6 +15,32 @@ def get_bin_path(bin_dir):
     return os.path.join(bin_dir, "bin", subdirs[system])
 
 
+def _runnable_native(path: str) -> bool:
+    """True when a native binary is a runnable file, restoring a lost exec bit when possible.
+
+    Why: a 0644 binary wired as an absolute path fails EACCES at Popen with the PATH
+    fallback disabled. Mirrors the tool_registry.installers repair (not imported from
```
Comment (Svilen-Stefanov) [inline vscode_constants.py:22]: The comment here suggest that we're trying to avoid some circular dependency. If that's the case, we should just extract that into a helper that both of the files are calling.

**Review by Svilen-Stefanov — COMMENTED** [review]
It would be great if we could reproduce this first and then make sure that the changes here indeed fix it.
